visio_ml/src/modules/compile_results.py

```diff
@@ -0,0 +1,49 @@
+import cv2
+import firebase_admin
+import datetime
+
+class ResultCompiler:
+    PATH_TO_CREDENTIALS = "" # TODO: Add file path to firebase credentials
+
+    # results_db_path: URL to the realtime database for results
+    # results_destination_path: destination file in firebase storage for results
+    # video_db_name: name of the realtime database for videos
+    def __init__(self, exercise, body_part, results_db_path, results_destination_path, video_db_name, frame_rate):
+        self.cred = firebase_admin.credentials.Certificate(self.PATH_TO_CREDENTIALS)
+        self.exercise = exercise
+        self.body_part = body_part
+        self.results_db_path = results_db_path
+        self.results_destination_path = results_destination_path
+        self.video_db_name = video_db_name
+        self.results = []
+        self.total = 0
+        self.video_file_name = f"{exercise}_{body_part}_{frame_rate}fps_{datetime.now()}.avi"
+        self.out = cv2.VideoWriter(self.video_file_name, cv2.VideoWriter_fourcc('M','J','P','G'), frame_rate, (1152, 648))
+    
+    def record_angle(self, angle) -> None:
```

Comment:
the argument angle will be a dictionary {'left': 100, 'right': 90}

Reply:
So `body_part` in the constructor won't be `left_shoulder`, it would just be `shoulder` etc.?

Reply:
yup